contracts/settling_game/L07_Crypts.cairo

```diff
@@ -0,0 +1,196 @@
+# ____MODULE_L07___CRYPTS_LOGIC
+#   Staking/Unstaking a crypt.
+#
+# MIT License
+
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin, BitwiseBuiltin
+from starkware.starknet.common.syscalls import (
+    get_caller_address,
+    get_block_timestamp,
+    get_contract_address,
+)
+from starkware.cairo.common.uint256 import Uint256
+
+from contracts.settling_game.utils.game_structs import ModuleIds, ExternalContractIds, RealmData
+from contracts.settling_game.utils.constants import TRUE, FALSE
+from contracts.settling_game.library.library_module import (
+    MODULE_controller_address,
+    MODULE_only_approved,
+    MODULE_initializer,
+)
+
+from openzeppelin.token.erc721.interfaces.IERC721 import IERC721
+from contracts.settling_game.interfaces.crypts_IERC721 import crypts_IERC721
+from contracts.settling_game.interfaces.s_crypts_IERC721 import s_crypts_IERC721
+from contracts.settling_game.interfaces.imodules import IModuleController, IL08_Crypts_Resources
+
+from openzeppelin.upgrades.library import (
+    Proxy_initializer,
+    Proxy_only_admin,
+    Proxy_set_implementation,
+)
+
+##########
+# EVENTS #
+##########
+
+# Staked = 🗝️ unlocked
+# Unstaked = 🔒 locked (because Lore ofc)
+
+@event
+func Settled(owner : felt, token_id : Uint256):
+end
+
+@event
+func UnSettled(owner : felt, token_id : Uint256):
+end
+
+###########
+# STORAGE #
+###########
+
+# STAKE TIME - This is used as the main identifier for staking time
+# It is updated on Resource Claim, Stake, Unstake
+@storage_var
+func time_staked(token_id : Uint256) -> (time : felt):
+end
+
+###############
+# CONSTRUCTOR #
+###############
+
+@external
+func initializer{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    address_of_controller : felt, proxy_admin : felt
+):
+    MODULE_initializer(address_of_controller)
+    Proxy_initializer(proxy_admin)
+    return ()
+end
+
+@external
+func upgrade{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    new_implementation : felt
+):
+    Proxy_only_admin()
+    Proxy_set_implementation(new_implementation)
+    return ()
+end
+
+############
+# EXTERNAL #
+############
+
+# SETTLES CRYPT
+@external
+func settle{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    token_id : Uint256
+) -> (success : felt):
+    alloc_locals
+    let (caller) = get_caller_address()
```

Comment:
I'm not sure how to fix this one. I could wrap it in an 'if' but not sure what that would do to help. This just seems like a starknet quirk that's throwing a warning

Reply:
we can leave this one, I don't think it impacts anything